core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -180,12 +182,19 @@ public void onNext(HttpObject o) {
                 break;
             }
             case NEEDS_TRAILERS: {
-                if (o instanceof HttpData || o instanceof ResponseHeaders) {
+                if (o instanceof ResponseHeaders) {
                     failAndRespond(new IllegalStateException(
-                            "published an HttpData or a ResponseHeaders: " + o +
+                            "published a ResponseHeaders: " + o +
                             " (expected: an HTTP trailers). service: " + service()));
                     return;
                 }
+                if (o instanceof HttpData) {
```

Comment:
Just wondering is this change enough to fix the issue?

Reply:
I think so. Instead of resetting, we can just ignore the content and call `subscription.request(1)` so that the response is completed in `onComplete`.